specs/conductor.md

```diff
@@ -1,59 +1,137 @@
+
+When blocks are received from the gossip network, their transactions are
+filtered based on the rollup's namespace, then are sent to the rollup for
+execution. The execution hash that is returned from the rollup is then stored in
+a hash map for Sequencer block hash -> execution hash.
+
+When blocks are received from the DA layer, the hash map of Sequencer block hash
+-> execution hash is checked to see if the block has already passed through the
+Conductor from the Sequencer. If the block isn't seen, it is filtered and sent
+to the rollup for execution exactly the same way the transactions are sent when
+received from the Sequencer, then a message to finalize the block is sent.
+If the block is already present in the hash map, just the finalize block message
+is sent. After being finalized, the Sequencer block hash -> execution hash entry
+in the hash map is deleted.
 
 ## Execution Data
+
 ### Transaction Filtering
-The Conductor is designed so that for each rollup node running, an instance of the Conductor will be run with it.
-The `chain_id` that Conductor uses for the rollup namespace is pulled from the rollup instance's config. As such, the Conductor is only aware of the single namespace of the rollup it is supporting. The `chain_id`/namespace is tracked in the sequencer block that contains the transactions for the rollup and not in the transactions themselves. When a sequencer block is received from the gossip network the Conductor only pulls out the transactions for its namespace and passes those transaction, and the previous execution hash from the rollup to the execution layer. See the [astria execution api](https://github.com/astriaorg/astria/blob/main/specs/execution-api.md) for more details.
+
+An instance of the Conductor is meant to be run alongside the rollup node.
+
+The `chain_id` that Conductor uses as the rollup namespace is pulled from the rollup's config. As such, the Conductor is only aware of the single namespace of the rollup it is supporting. The `chain_id` or namespace is tracked in the Sequencer block that contains the transactions for the rollup and not in the transactions themselves.
+
+When a Sequencer block is received from the gossip network the Conductor filters the transactions for its namespace and passes only those transaction, and the previous execution hash from the rollup to the execution layer. See the [astria execution api](https://github.com/astriaorg/astria/blob/main/specs/execution-api.md) for more details.
+
+### Data Validation
+
+Data is validated before being sent to the rollup for execution. Validation occurs in two places:
+
+- When blocks are received from the gossip network, the data is validated in Driver's `handle_network_event` using `BlockVerifier::validate_sequencer_block_data`
+- When blocks are fetched from the DA layer, the data is validated in Reader's `get_new_blocks` using `BlockVerifier::validate_signed_namespace_data` and `BlockVerifier::validate_rollup_data`
```

Comment:
this changes after  #290 actually

Reply:
I'll create another branch to capture the changes for this!

Reply:
Created https://github.com/astriaorg/astria/pull/322